src/tests/ftest/util/general_utils.py

```diff
@@ -288,28 +288,104 @@ def run_command(command, timeout=60, verbose=True, raise_exception=True,
         timeout (int, optional): command timeout in seconds. Defaults to None.
+        verbose (bool, optional): display message for command execution. Defaults to False.
 
     Returns:
         Task: a ClusterShell.Task.Task object for the executed command
 
     """
+    if not isinstance(hosts, NodeSet):
+        hosts = NodeSet.fromlist(hosts)
     task = task_self()
     # Enable forwarding of the ssh authentication agent connection
     task.set_info("ssh_options", "-oForwardAgent=yes")
-    kwargs = {"command": command, "nodes": NodeSet.fromlist(hosts)}
+    kwargs = {"command": command, "nodes": hosts}
     if timeout is not None:
         kwargs["timeout"] = timeout
+    if verbose:
+        log = getLogger()
+        log.info("Running on %s: %s", hosts, command)
     task.run(**kwargs)
     return task
 
 
+def check_task(task, logger=None):
+    """Check the results of the executed task and get the output.
+
+    Args:
+        task (Task): a ClusterShell.Task.Task object for the executed command
+
+    Returns:
+        bool: if the command returned an 0 exit status on every host
+
+    """
+    def check_task_log(message):
+        """Log the provided text if a logger is present.
+
+        Args:
+            message (str): text to display
+        """
+        if logger:
+            logger.info(message)
+
+    # Create a dictionary of hosts for each unique return code
+    results = dict(task.iter_retcodes())
+
+    # Display the command output
+    for code in sorted(results):
+        output_data = list(task.iter_buffers(results[code]))
+        if not output_data:
+            output_data = [["<NONE>", results[code]]]
+        for output, o_hosts in output_data:
+            node_set = NodeSet.fromlist(o_hosts)
+            lines = list(output.splitlines())
+            if len(lines) > 1:
+                # Print the sub-header for multiple lines of output
+                check_task_log("    {}: rc={}, output:".format(node_set, code))
+            for number, line in enumerate(lines):
+                if isinstance(line, bytes):
+                    line = line.decode("utf-8")
+                if len(lines) == 1:
+                    # Print the sub-header and line for one line of output
+                    check_task_log("    {}: rc={}, output: {}".format(node_set, code, line))
+                    continue
+                try:
+                    check_task_log("      {}".format(line))
+                except IOError:
+                    # DAOS-5781 Jenkins doesn't like receiving large amounts of data in a short
+                    # space of time so catch this and retry.
+                    check_task_log(
+                        "*** DAOS-5781: Handling IOError detected while processing line {}/{} with "
+                        "retry ***".format(*number + 1, len(lines)))
+                    time.sleep(5)
+                    check_task_log("      {}".format(line))
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string